packages/lib/onedrive-api.ts

```diff
@@ -339,6 +341,15 @@ export default class OneDriveApi {
 
 					await handleRequestRepeat(error);
 					continue;
+				} else if (error?.code === 'activityLimitReached' && response?.headers?._headers['retry-after'][0]) {
+					// Wait for OneDrive throttling
+					// Relavent Microsoft Docs: https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#best-practices-to-handle-throttling
+					// Decrement retry count as multiple sync threads will cause repeated throttling errors - this will wait until throttling is resolved to continue, preventing a hard stop on the sync
+					i--;
+					const sleep_seconds = response.headers._headers['retry-after'][0];
```

Comment:
sleepSeconds.

Could you also convert it to a number using `Number(response.headers._headers['retry-after'][0])` and perhaps also validate with `isNaN` just to be sure. If it's NaN (very unlikely), maybe just throw an error.

Reply:
Converted to Number when passing to function and test for isNaN() in if statement

```javascript
else if (error?.code === 'activityLimitReached' && response?.headers?._headers['retry-after'][0] && !isNaN(Number(response?.headers?._headers['retry-after'][0]))) {
  // Wait for OneDrive throttling
  // Relavent Microsoft Docs: https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#best-practices-to-handle-throttling
  // Decrement retry count as multiple sync threads will cause repeated throttling errors - this will wait until throttling is resolved to continue, preventing a hard stop on the sync
  i--;
  const sleepSeconds = response.headers._headers['retry-after'][0];
  logger.info(`OneDrive Throttle, sync thread sleeping for ${sleepSeconds} seconds...`);
  await handleRequestRepeat(error, Number(sleepSeconds));
  continue;
  }
```